File: algorithms/python/217-contains-duplicate.py
# https://leetcode.com/problems/contains-duplicate/
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = dict()
        for num in nums:
            if num in seen.keys():
                return True
            seen[num] = 1
        return False

File: algorithms/c++/217-contains-duplicate.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); i++) {
            if (seen.find(nums[i]) != seen.end()) {
                return true;
            }
            seen[nums[i]] = 1;
        }
        return false;
    }
}